crates/kitsune_p2p/proxy/src/tx2.rs

```diff
@@ -144,7 +156,57 @@ fn promote_addr(base_addr: &TxUrl, cert: &Tx2Cert) -> KitsuneResult<TxUrl> {
         .into())
 }
 
-#[allow(dead_code)]
+#[derive(Clone)]
+struct Backoff {
```

Comment:
probably don't need to do this now, but it might be nice to have the basic math and struct behind the backoff as something that people using the scheduler could build on top of as well

Reply:
Yeah, at this point I think we have a couple structs like this tailored for the specific situations scattered across the codebase.